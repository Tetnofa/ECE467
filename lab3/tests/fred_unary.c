void f1() {
  int x;
  int y;
  bool b1;
  bool b2;
  
  -5;
  -x;
  -(-10);
  -(y + 3);

  !true;
  !b1;
  !(false || true);
  !(!b2);
  !(b1 && b2);
  !((b1 || false) && (b2 || true));

  !5;          // Invalid: '!' applied to int instead of bool
  -x &&y;      // Invalid: '-' should apply to entire integer expression
  -(true);     // Invalid: '-' applied to bool instead of int
  -(!b1);      // Invalid: '-' applied to result of boolean expression
  -true;       // Invalid: '-' applied to a bool constant
  !(x + y);    // Invalid: '!' applied to int expression
  !(-x);       // Invalid: '!' applied to result of integer expression
  -(b1 || b2); // Invalid: '-' applied to boolean expression
}
