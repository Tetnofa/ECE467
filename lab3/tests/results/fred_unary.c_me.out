sema: 19:3 : type mismatch 
sema: 20:2 : type mismatch 
sema: 21:4 : type mismatch 
sema: 22:4 : type mismatch 
sema: 23:3 : type mismatch 
sema: 24:4 : type mismatch 
sema: 25:4 : type mismatch 
sema: 26:4 : type mismatch 
